### PR TITLE
refactor: use flag api to define configs in pegasus_reporter

### PR DESCRIPTION
**Pull request by neverchanje**
### What problem does this PR solve? 

This PR redefines the configurations in pegasus_reporter module using flag API. No logical changes.

### Check List 

Related changes

- Need to cherry-pick to the release branch

### CODE REVIEW

File: src/reporter/pegasus_counter_reporter.cpp [hunk 1]
```diff
@@ -2,28 +2,47 @@
...
+DSN_DEFINE_string("pegasus.server", falcon_host, "127.0.0.1", "falcon agent host");
+DSN_DEFINE_uint64("pegasus.server", falcon_port, 1988, "falcon agent port");
+DSN_DEFINE_string("pegasus.server", falcon_path, "/v1/push", "falcon http path");
+
+DSN_DEFINE_uint64("pegasus.server",
+                  perf_counter_update_interval_seconds,
+                  10,
+                  "perf_counter_update_interval_seconds");
+DSN_DEFINE_bool("pegasus.server", perf_counter_enable_logging, true, "perf_counter_enable_logging");
+DSN_DEFINE_string("pegasus.server", perf_counter_sink, "", "perf_counter_sink");
```
Comment (acelyc111) [inline src/reporter/pegasus_counter_reporter.cpp:41]: Reorder as top-down, or dict order by config key would be better.

File: src/reporter/pegasus_counter_reporter.cpp [hunk 2]
```diff
@@ -2,28 +2,47 @@
...
-#include <dsn/dist/fmt_logging.h>
+#include "pegasus_counter_reporter.h"
 
 using namespace ::dsn;
 
+DSN_DEFINE_uint64("pegasus.server", prometheus_port, 9091, "prometheus exposer port");
+
+DSN_DEFINE_string("pegasus.server", falcon_host, "127.0.0.1", "falcon agent host");
+DSN_DEFINE_uint64("pegasus.server", falcon_port, 1988, "falcon agent port");
+DSN_DEFINE_string("pegasus.server", falcon_path, "/v1/push", "falcon http path");
```
Comment (acelyc111) [inline src/reporter/pegasus_counter_reporter.cpp:34]:
```suggestion
DSN_DEFINE_string("pegasus.server", falcon_path, "/v1/push", "falcon agent http path");
```

File: src/reporter/pegasus_counter_reporter.cpp (hunk 1 above)
Comment (neverchanje) [reply src/reporter/pegasus_counter_reporter.cpp:41]: I don't get your point. What's top-down order by config-key? Could you please write a suggestion showing that?

Comment (acelyc111) [reply src/reporter/pegasus_counter_reporter.cpp]: There are 3 types of config group, top-down order is:
- configs for server            # top level, we can select prometheus or falcon
- configs for prometheus  # lower level
- configs for falcon            # lower level
So when developers read code here, he know it's useless to config falcon items when select prometheus sink.

File: src/reporter/pegasus_counter_reporter.cpp [hunk 3]
```diff
@@ -2,28 +2,48 @@
...
 
 #include "base/pegasus_utils.h"
 #include "pegasus_io_service.h"
-
-#include <chrono>
-#include <map>
-#include <memory>
-#include <string>
-#include <dsn/dist/fmt_logging.h>
+#include "pegasus_counter_reporter.h"
```
Comment (acelyc111) [inline src/reporter/pegasus_counter_reporter.cpp:26]: The related header should be placed at first of all.

**Pull request merged**